src/character.cpp

```diff
@@ -1472,6 +1472,9 @@ bool Character::check_mount_will_move( const tripoint &dest_loc )
     if( !is_mounted() ) {
         return true;
     }
+    if( mounted_creature && mounted_creature->has_flag( MF_COMBAT_MOUNT ) ) {
+        return true;
+    }
```

Comment:
```suggestion
    if( !is_mounted() || mounted_creature->has_flag( MF_COMBAT_MOUNT ) ) {
        return true;
    }
```

This can be combined. And since `is_mounted` will check for `mounted_creature`, there's no need to check again.

Reply:
Thanks!